src/main/java/org/fcrepo/auth/webac/LinkHeaderProvider.java

```diff
@@ -0,0 +1,86 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.fcrepo.auth.webac;
+
+import static com.hp.hpl.jena.rdf.model.ModelFactory.createDefaultModel;
+import static org.fcrepo.auth.webac.URIConstants.WEBAC_ACCESS_CONTROL_VALUE;
+import static org.slf4j.LoggerFactory.getLogger;
+
+import javax.jcr.Session;
+import javax.ws.rs.core.UriInfo;
+
+import org.fcrepo.http.commons.api.UriAwareHttpHeaderFactory;
+import org.fcrepo.http.commons.session.SessionFactory;
+import org.fcrepo.kernel.api.identifiers.IdentifierConverter;
+import org.fcrepo.kernel.api.models.FedoraResource;
+import org.fcrepo.kernel.api.services.NodeService;
+import org.fcrepo.kernel.modeshape.rdf.impl.DefaultIdentifierTranslator;
+import org.fcrepo.kernel.modeshape.rdf.impl.PropertiesRdfContext;
+
+import org.slf4j.Logger;
+
+import com.google.common.collect.ArrayListMultimap;
+import com.google.common.collect.ListMultimap;
+import com.google.common.collect.Multimap;
+import com.hp.hpl.jena.rdf.model.Model;
+import com.hp.hpl.jena.rdf.model.Resource;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+/**
+ * Insert WebAC Link headers to responses
+ *
+ * @author whikloj
+ * @since 2015-10-30
+ */
+@Component
+public class LinkHeaderProvider implements UriAwareHttpHeaderFactory {
+
+    private static final Logger LOGGER = getLogger(LinkHeaderProvider.class);
+
+    @Autowired
+    private SessionFactory sessionFactory;
+
+    @Autowired
+    private NodeService nodeService;
+
+    @Override
+    public Multimap<String, String> createHttpHeadersForResource(final UriInfo uriInfo,
+            final FedoraResource resource) {
+
+        final Session internalSession = sessionFactory.getInternalSession();
+        final IdentifierConverter<Resource, FedoraResource> translator =
+                new DefaultIdentifierTranslator(internalSession);
+        final Model model = createDefaultModel();
+        final ListMultimap<String, String> headers = ArrayListMultimap.create();
+
+        LOGGER.debug("Adding WebAC Link Header for Resource: {}", resource);
+
+        nodeService.find(internalSession, resource.getPath()).getTriples(translator, PropertiesRdfContext.class)
+        .filter(t -> model.asStatement(t).getPredicate().hasURI(WEBAC_ACCESS_CONTROL_VALUE))
+        .filter(t -> t.getObject().isURI())
+        .forEachRemaining(t -> {
+            headers.put("Link", uriInfo.getBaseUriBuilder()
```

Comment:
Instead of "hand creating" the Link header, please use `javax.ws.rs.core.Link`, such as in:
https://github.com/fcrepo4/fcrepo4/blob/master/fcrepo-http-commons/src/main/java/org/fcrepo/http/commons/exceptionhandlers/ConstraintExceptionMapper.java#L47-L52